src/mlpack/core/cv/k_fold_cv.hpp

```diff
@@ -0,0 +1,219 @@
+/**
+ * @file k_fold_cv.hpp
+ * @author Kirill Mishchenko
+ *
+ * k-fold cross-validation.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_CV_K_FOLD_CV_HPP
+#define MLPACK_CORE_CV_K_FOLD_CV_HPP
+
+#include <mlpack/core/cv/meta_info_extractor.hpp>
+#include <mlpack/core/cv/cv_base.hpp>
+
+namespace mlpack {
+namespace cv {
+
+/**
+ * The class KFoldCV implements k-fold cross-validation for regression and
+ * classification algorithms.
+ *
+ * To construct a KFoldCV object you need to pass the k parameter and arguments
+ * that specify data. For the latter see the CVBase constructors as a reference
+ * - the CVBase constructors take exactly the same arguments as ones that are
+ * supposed to be passed after the k parameter in the KFoldCV constructor.
```

Comment:
I think this way of referring someone to the other arguments of the constructor
could be a little bit confusing, and because we are not exposing the CVBase
constructors directly to the user, we can't try to force Doxygen to print all of
the CVBase constructors directly along with the KFoldCV constructors.

The only solution I could see would be to replicate the four constructors of
CVBase here.  This would cause extra code duplication, but it would be for the
sake of clarity of documentation.  Would you be willing to do that?  I think
that that would also allow inlining of the Init() functions into the
constructors directly, so maybe the amount of extra code and functions would not
be that many...